web/packages/teleport/src/AccessRequests/utils.test.ts

```diff
@@ -0,0 +1,195 @@
+    const duration = {
+      ...defaultValues,
+      days,
+      hours,
+      minutes,
+    };
+    let d = new Date(currentDate);
+    d.setDate(currentDate.getDate() + days);
+    d.setHours(currentDate.getHours() + hours);
+    d.setMinutes(currentDate.getMinutes() + minutes);
+    const timestamp = d.getTime();
+    result.push({
+      timestamp,
+      duration,
+    });
+  }
+  return result;
+}
+
+describe('generate middle times', () => {
+  const cases: {
+    name: string;
+    created: string;
+    sessionTTL: string;
+    maxDuration: string;
+    expected: Array<{
+      days: number;
+      hours: number;
+      minutes: number;
+    }>;
+  }[] = [
+    {
+      name: '3 days max',
+      created: '2021-09-01T00:00:00.000Z',
+      sessionTTL: '2021-09-01T01:00:00.000Z',
+      maxDuration: '2021-09-04T00:00:00.000Z',
+      expected: [
+        {
+          days: 0,
+          hours: 1,
+          minutes: 0,
+        },
+        {
+          days: 1,
+          hours: 0,
+          minutes: 0,
+        },
+        {
+          days: 2,
+          hours: 0,
+          minutes: 0,
+        },
+        {
+          days: 3,
+          hours: 0,
+          minutes: 0,
+        },
+      ],
+    },
+    {
+      name: '1 day max',
+      created: '2021-09-01T00:00:00.000Z',
+      sessionTTL: '2021-09-01T01:00:00.000Z',
+      maxDuration: '2021-09-02T00:00:00.000Z',
+      expected: [
+        {
+          days: 0,
+          hours: 1,
+          minutes: 0,
+        },
+        {
+          days: 1,
+          hours: 0,
+          minutes: 0,
+        },
+      ],
+    },
+    {
+      name: 'session ttl is 10 min',
+      created: '2021-09-01T00:00:00.000Z',
+      sessionTTL: '2021-09-01T00:10:00.000Z',
+      maxDuration: '2021-09-03T00:00:00.000Z',
+      expected: [
+        {
+          days: 0,
+          hours: 0,
+          minutes: 10,
+        },
+        {
+          days: 1,
+          hours: 0,
+          minutes: 0,
+        },
+        {
+          days: 2,
+          hours: 0,
+          minutes: 0,
+        },
+      ],
+    },
+    {
+      name: '10 minutes min - real values',
+      created: '2023-09-21T20:50:52.669012121Z',
+      sessionTTL: '2023-09-21T21:00:52.669081473Z',
+      maxDuration: '2023-09-27T20:50:52.669081473Z',
+      expected: [
+        {
+          days: 0,
+          hours: 0,
+          minutes: 10,
+        },
+        {
+          days: 1,
+          hours: 0,
+          minutes: 0,
+        },
+        {
+          days: 2,
+          hours: 0,
+          minutes: 0,
+        },
+        {
+          days: 3,
+          hours: 0,
+          minutes: 0,
+        },
+        {
+          days: 4,
+          hours: 0,
+          minutes: 0,
+        },
+        {
+          days: 5,
+          hours: 0,
+          minutes: 0,
+        },
+        {
+          days: 6,
+          hours: 0,
+          minutes: 0,
+        },
+      ],
+    },
+  ];
+
+  for (let tc of cases) {
```

Comment:
The proper way to do this with Jest would be a `describe.each` block (similar to Go subtests): https://jestjs.io/docs/api#describeeachtablename-fn-timeout

Reply:
I've tried to use it, but I had some issues. I'll try again tho 😅